c3.php

```diff
@@ -8,420 +8,463 @@
-$configFile = realpath(__DIR__) . DIRECTORY_SEPARATOR . 'codeception.yml';
+            // fix for improperly encoded JSON in Code Coverage cookie with WebDriver.
+            // @see https://github.com/Codeception/Codeception/issues/874
+            if (!is_array($cookie)) {
+                $cookie = json_decode($cookie, true);
+            }
 
-if (isset($_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE_CONFIG'])) {
-    $configFile = realpath(__DIR__) . DIRECTORY_SEPARATOR . $_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE_CONFIG'];
-}
-if (file_exists($configFile)) {
-    // Use codeception.yml for configuration.
-} elseif (file_exists($configDistFile)) {
-    // Use codeception.dist.yml for configuration.
-    $configFile = $configDistFile;
-} else {
-    __c3_error(sprintf("Codeception config file '%s' not found", $configFile));
-}
-try {
-    \Codeception\Configuration::config($configFile);
-} catch (\Exception $e) {
-    __c3_error($e->getMessage());
-}
+            if ($cookie) {
+                foreach ($cookie as $key => $value) {
+                    if (!empty($value)) {
+                        $_SERVER["HTTP_X_CODECEPTION_" . strtoupper($key)] = $value;
+                    }
+                }
+            }
+        }
 
-if (!defined('C3_CODECOVERAGE_MEDIATE_STORAGE')) {
+        if (!array_key_exists('HTTP_X_CODECEPTION_CODECOVERAGE', $_SERVER)) {
+            return;
+        }
 
-    // workaround for 'zend_mm_heap corrupted' problem
-    gc_disable();
+        if (!function_exists('__c3_error')) {
+            function __c3_error($message)
+            {
+                $errorLogFile = defined('C3_CODECOVERAGE_ERROR_LOG_FILE') ?
+                    C3_CODECOVERAGE_ERROR_LOG_FILE :
+                    C3_CODECOVERAGE_MEDIATE_STORAGE . DIRECTORY_SEPARATOR . 'error.txt';
+                if (is_writable($errorLogFile)) {
+                    file_put_contents($errorLogFile, $message);
+                } else {
+                    $message = "Could not write error to log file ($errorLogFile), original message: $message";
+                }
+                if (!headers_sent()) {
+                    header('X-Codeception-CodeCoverage-Error: ' . str_replace("\n", ' ', $message), true, 500);
+                    setcookie('CODECEPTION_CODECOVERAGE_ERROR', $message);
+                }
+            }
+        }
 
-    $memoryLimit = ini_get('memory_limit');
-    $requiredMemory = '384M';
-    if ((substr($memoryLimit, -1) === 'M' && (int)$memoryLimit < (int)$requiredMemory)
-        || (substr($memoryLimit, -1) === 'K' && (int)$memoryLimit < (int)$requiredMemory * 1024)
-        || (ctype_digit($memoryLimit) && (int)$memoryLimit < (int)$requiredMemory * 1024 * 1024)
-    ) {
-        ini_set('memory_limit', $requiredMemory);
-    }
+        // Autoload Codeception classes
+        if (!class_exists('\\Codeception\\Codecept') || !function_exists('codecept_is_path_absolute')) {
```

Comment:
Use `::class` (and add import at top)
```suggestion
        if (!class_exists(Codecept::class) || !function_exists('codecept_is_path_absolute')) {
```